homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,349 @@
+
+        mode = HVACMode.OFF
+        if self.coordinator.get_coil_value(self._coil_use_room_sensor) == "ON":
+            if (
+                self.coordinator.get_coil_value(self._coil_cooling_with_room_sensor)
+                == "ON"
+            ):
+                mode = HVACMode.HEAT_COOL
+            else:
+                mode = HVACMode.HEAT
+        self._attr_hvac_mode = mode
+
+        setpoint_heat = self.coordinator.get_coil_float(self._coil_setpoint_heat)
+        setpoint_cool = self.coordinator.get_coil_float(self._coil_setpoint_cool)
+
+        if mode == HVACMode.HEAT_COOL:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = setpoint_heat
+            self._attr_target_temperature_high = setpoint_cool
+        elif mode == HVACMode.HEAT:
+            self._attr_target_temperature = setpoint_heat
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+        else:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+
+        if prio := self.coordinator.get_coil_value(self._coil_prio):
+            if (
+                mixing_valve_state := self.coordinator.get_coil_value(
+                    self._coil_mixing_valve_state
+                )
+            ) and mixing_valve_state == 30:
+                self._attr_hvac_action = HVACAction.IDLE
+            elif prio == "HEAT":
+                self._attr_hvac_action = HVACAction.HEATING
+            elif prio == "COOLING":
+                self._attr_hvac_action = HVACAction.COOLING
+            else:
+                self._attr_hvac_action = HVACAction.IDLE
+        else:
+            self._attr_hvac_action = None
+
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        if not self._coil_active_accessory_address:
+            return True
+
+        if active_accessory := self.coordinator.get_coil_value(
+            self._coil_active_accessory_address
+        ):
+            return active_accessory == "ON"
+
+        return False
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set target temperatures."""
+
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE, self._attr_hvac_mode)
+
+        if (temperature := kwargs.get(ATTR_TEMPERATURE, None)) is not None:
+            if hvac_mode == HVACMode.HEAT:
+                await self.coordinator.async_write_coil(
+                    self._coil_setpoint_heat, temperature
+                )
+            elif hvac_mode == HVACMode.COOL:
+                await self.coordinator.async_write_coil(
+                    self._coil_setpoint_cool, temperature
+                )
+            else:
+                raise HomeAssistantError(
```

Comment:
> Operations like service calls and entity methods (e.g. Set HVAC Mode) have proper exception handling. Raise ValueError on invalid user input and raise HomeAssistantError for other failures such as a problem communicating with a device.

https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=quality#silver-

I think this should be `ValueError`